starter-code/app.js

```diff
@@ -2,4 +2,67 @@ const express = require('express');
 const app = express();
 const Chuck  = require('chucknorris-io');
 const client = new Chuck();
+const bodyParser = require('body-parser');
 
+app.use(express.static('public'));
+app.set('views', __dirname + '/views');
+app.set('view engine', 'ejs');
+app.use(bodyParser.urlencoded({ extended: true }));
+
+app.get('/', (req, res, next) => {
+  res.render('home');
+});
+
+app.get('/random', (req, res, next) => {
+
+  client.getRandomJoke()
+    .then((response) => {
+      console.log(response);
+      res.render('joke-by-category', {joke: response});
+    }).catch((err) => {
+      // handle error
+    });
+});
+
+app.get('/categories', (req, res, next) => {
+  if (req.query.cat === undefined){
+    client.getJokeCategories()
+    .then((response)=>  {
+      res.render('categories', {categories: response})
```

Comment:
You could also write like this to make it shorter: 

 client.getJokeCategories().then(function (categories) {
    res.render('categories', {categories});

then you don't need to do "{categories: response}"